daffodil-core/src/main/scala/org/apache/daffodil/dsom/SequenceGroup.scala

```diff
@@ -147,6 +151,18 @@ abstract class SequenceGroupTermBase(
     }
   }.value
 
+  /**
+   * Provides validation for assert and discriminator placement
+   */
+  protected final lazy val checkIfNonEmptyAndDiscrimsOrAsserts: Unit = {
+    val msg = "Counterintuitive placement detected. Wrap the discriminator or assert " +
+      "in an empty sequence to evaluate before the contents.\n"
```

Comment:
How does the \n help readability?  Just curious (I grep'ed the source for "SDW(WarnID" and only 3 out of 25 SDW's had \n in their messages).

Reply:
Yeah, in general we shouldn't have newlines in errors. The code the prints out diagnostics is responsible for that. There are some cases where a parse error includes multiple lines of information so newlines can help make that a little more easier to read. But usually it should be avoided. In this case, it isn't needed.

Reply:
Looks like you still need to remove the \n.